feature-dapp-impl/src/main/java/io/novafoundation/nova/feature_dapp_impl/utils/tabs/RealBrowserTabService.kt

```diff
@@ -118,4 +129,27 @@ class RealBrowserTabService(
             activeSessions[tabId] ?: addNewSession(tab)
         )
     }
+
+    override fun destroyActiveSessions() {
+        activeSessions.removeAll()
+    }
+
+    /*
+    We should update page title and url each time page is changed to have correct tab state in persistent storage
+     */
+    override fun onPageChanged(tabId: String, url: String, title: String?) {
+        launch(Dispatchers.Main) {
+            withContext(Dispatchers.Default) { browserTabInternalRepository.changeCurrentUrl(tabId, url) }
+            makeTabSnapshot(tabId)
+        }
+    }
+
+    private suspend fun makeTabSnapshot(tabId: String) {
+        val pageSession = activeSessions.get(tabId)
+
+        if (pageSession != null) {
+            val snapshot = pageSnapshotBuilder.getPageSnapshot(pageSession)
```

Comment:
Only this needs Main dispatcher right? I think we should just switch it here instead of switching multple times to Default

Reply:
No, it's important to run it not in coroutine since sometimes we run it in pause and before onDestroyView. So If view destroyed during making a snapshot the snapshot may be drawn partly